### PR TITLE
Testing fit with experimental data

### PR DESCRIPTION
**Pull request by yfukai**
Attempt to make the test with experimental data using `pooch`. Assume #53 to be approved.

### CODE REVIEW

**Comment by Nicholas-Schaub** [comment]
Is this still WIP?

**Comment by yfukai** [comment]
Yes, this need to wait for #57 .

**Comment by yfukai** [comment]
This will close #52.

**Review by Nicholas-Schaub — CHANGES_REQUESTED** [review]
I think this looks good. However, I don't think we should move the inexact_alm to `_deprecated`, and I think we should keep the original code under `fit`. We had previously discussed keeping these for the purposes of reproducing the original results in the paper. If we could maybe place these in a separate section, and then have another setting to select this in addition to `ladmap` and `approximate`, I think that would be best.

File: src/basicpy/basicpy.py [hunk 1]
```diff
@@ -46,22 +54,28 @@
...
-
-
 class Device(Enum):
 
     cpu: str = "cpu"
     gpu: str = "gpu"
     tpu: str = "tpu"
 
 
+class FittingMode(Enum):
```
[the remark this answers is not in the document]
Comment (yfukai) [reply src/basicpy/basicpy.py:64]: Thanks! It seems the link is not working. If you wouldn't mind, can you paste the spnippet of the code?

**Comment by yfukai** [comment]
Closing as we merge PRs to #62.

**Comment by yfukai** [comment]
Hi, thanks @Nicholas-Schaub for your comment! Actually, the `approximate` fitting method is identical to the original `inexact_alm` (at https://github.com/yfukai/BaSiCpy/tree/reference_implementation/) except for the point that it's implemented in JAX. I compared them in https://github.com/yfukai/BaSiCpy/blob/3d_dct/misc_notebooks/compare_with_reference_impl.ipynb to find the results are close. In my opinion, I think it suffices to keep https://github.com/yfukai/BaSiCpy/tree/reference_implementation/ in the repo and we don't need to maintain the original impl. code (which is basically a copy of Matlab impl. to Python), but let's discuss this tomorrow!

**Comment by yfukai** [comment]
If it's really necessary I could write the `numpy` impl. of the approximate version, which should give exactly the same results as those of the original ones (but the exact agreement w/ the Matlab version is not guaranteed).

**Pull request closed without merging**